File: python/twoNumberSum.py

def twoNumbersSum(array, targetSum):
	array.sort()
	#print(array)
	left = 0
	right = len(array) - 1
	while left < right:
		currentSum = array[left] + array[right]
		if currentSum == targetSum:
			return [array[left], array[right]]
		elif currentSum < targetSum:
    			left += 1
		elif currentSum > targetSum:
			right -= 1
	return []

print(twoNumbersSum([1,2,5,4],7))

File: javascript/twoNumberSum.js

function twoNumberSum(array, target) {
  array.sort((a,b)=>( a - b ))
  let left = 0
  let right = array.length - 1
  let sum = 0
  while (left < right) {
    sum = array[left] + array[right]
    if ( sum === target) { return [array[left], array[right]] }
    else if ( sum < target ) { left += 1 }
    else if ( sum > target ) { right -= 1 }
  }

  return []
}

exports.twoNumberSum = twoNumberSum